crates/sui-core/src/authority_active/checkpoint_driver/mod.rs

```diff
@@ -548,7 +572,13 @@ where
                 info: AuthorityCheckpointInfo::Past(AuthenticatedCheckpoint::Certified(past)),
                 detail,
             }) => {
-                return Ok((past, detail));
+                return match (contents, &detail) {
+                    // TODO: this should be done by SafeClient
+                    (true, None) => Err(SuiError::ByzantineAuthoritySuspicion {
```

Comment:
This validator may not have the requested checkpoint, so it's not necessarily byzantine if we don't get it.

Reply:
That's not what this is checking - this is checking that if we specified `contents = true`, it sent us the contents, which allows my unwrap above to be safe.